### PR TITLE
files: add upload file support

### PR DESCRIPTION
**Pull request by jma**
* Adds files list and file uploads on the professionnal views.
* Adds icons depending of the mime types.

Need: https://github.com/rero/ng-core/pull/601

### CODE REVIEW

File: projects/admin/src/app/routes/route-tool.service.ts [hunk 1]
```diff
@@ -246,6 +246,7 @@ export class RouteToolService {
    *      - canDelete permission
    */
   permissions( record: any, recordType: string, membership = false): Observable<any> {
+    console.log(record, recordType);
```
Comment (Garfield-fr) [inline projects/admin/src/app/routes/route-tool.service.ts:249]: remove this

File: projects/admin/src/app/service/resources-files.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { TestBed } from '@angular/core/testing';
```
Comment (Garfield-fr) [inline projects/admin/src/app/service/resources-files.service.spec.ts:1]: missing copyright

File: projects/admin/src/app/service/resources-files.service.ts [hunk 3]
```diff
@@ -0,0 +1,284 @@
+import { HttpClient, HttpResponse } from "@angular/common/http";
```
Comment (Garfield-fr) [inline projects/admin/src/app/service/resources-files.service.ts:1]: missing copyright

File: projects/shared/src/lib/modules/prime-ng-import/prime-ng-import.module.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 /*
  * RERO ILS UI
- * Copyright (C) 2019-2024 RERO
+ * Copyright (C) 2019-2022 RERO
```
Comment (Garfield-fr) [inline projects/shared/src/lib/modules/prime-ng-import/prime-ng-import.module.ts:3]: 2024

**Review by Garfield-fr — CHANGES_REQUESTED** [review]
Please reformat your code with minimun 120 chars.

**Review by lauren-d — APPROVED** [review]

**Pull request merged**